### ISSUE TITLE
Constant calls to 'SET_ACTIVE_TOOL_GROUP' causing unwanted renders

### ISSUE THREAD

**Issue by henry-chris** [PDFTron/webviewer-ui#612]
V. 6.2.4 - 6.3.0  (and previous)

In `onToolModeUpdated.js` -

I believe that when mousing over different parts of a PDF CoreControls listens for mouse movement and then sends a call to this function when certain things are moused over.  If the newTool is the defaultTool then it sends a couple dispatches.  The main one is `setActiveToolGroup('')`.  In most cases the active toolgroup isn't actually being changed, but it may trigger re-renders many times when mousing over certain PDFs.

Suggest to not trigger a change when the 'activeToolGroup' doesn't need to change.

**Comment by henry-chris** [comment]
I would guess that a simple solution would be: (in viewerReducer)

```
    case 'SET_ACTIVE_TOOL_GROUP':
      if (payload.toolGroup === state.activeToolGroup) {
        return state;
      }

      return {
        ...state,
        activeToolGroup: payload.toolGroup,
      };
```

Could be wrong, though.

**Comment by ZhijieZhang** [comment]
Thanks for bringing this up.

I just fixed this in both `master` and `6.3`.

**Comment by henry-chris** [comment]
@ZhijieZhang  I'll leave this closed as that did help in some cases.  However, after further inspection I ended up doing this to prevent the Notespanel from being constantly re-rendered when mousing over annotations.

However, we have a custom Notespanel so I don't know if the same thing constantly causes the stock panel to re-render.  Obviously I had to pass `store` into the event-listener, also.

 ```
 if (newTool && newTool.name === defaultTool) {
    // stop viewer state re-renders
    if (selectors.getActiveToolGroup(getState()) !== '') {
      dispatch(actions.setActiveToolGroup(''));
    }
    dispatch(actions.closeElement('toolsOverlay'));
  }
```
